File: soyeong/programmers-2/튜플.py
def solution(s):
    answer = []
    # 문자열을 숫자만 남기기 위해 1차 단계 문자열 앞뒤 {{}}를 제거
    step1_s = s[2:-2]
    # 문자열을 숫자만 남기기 위해 2차 단계로 숫자 사이의 문자열 },{ 기준으로 배열 만들기
    step2_s = step1_s.split("},{")
    # {{a1}, {a1, a2}, {a1, a2, a3}, {a1, a2, a3, a4}, ... {a1, a2, a3, a4, ..., an}}
    # 뒤섞여있는 집합을 튜플로 복원하기 위해 이런 식으로 정렬해야 하기 때문에 배열안의 길이로 정렬
    step2_s.sort(key=len)

    for one in step2_s:
        # 각 집합의 원소들로 나눔
        num_arr = one.split(",")
        for num in num_arr:
            # 숫자가 answer에 없다면 포함
            if int(num) not in answer:
                answer.append(int(num))

    return answer

File: kyulee/programmers-2/튜플.java
import java.util.*;

class 튜플 {
    public int[] solution(String s) {
        // 문자열 파싱을 위한 전처리
        s = s.substring(2, s.length() - 2).replace("},{", "-");

        // '-'를 기준으로 문자열을 나누어 배열에 저장
        String[] tuples = s.split("-");

        // 길이순으로 정렬
        Arrays.sort(tuples, Comparator.comparingInt(String::length));

        List<Integer> answerList = new ArrayList<>();

        for (String tuple : tuples) {
            String[] elements = tuple.split(",");

            for (String element : elements) {
                int num = Integer.parseInt(element);

                if (!answerList.contains(num)) {
                    answerList.add(num);
                }
            }
        }

        // 리스트를 배열로 변환
        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }

        return answer;
    }
}